storage/engine/db.cc

```diff
@@ -492,6 +492,54 @@ bool MergeTimeSeriesValues(cockroach::proto::Value *left, const cockroach::proto
     return true;
 }
 
+// ConsolidateTimeSeriesValue processes a single value which contain
+// InternalTimeSeriesData messages. This method will sort the sample collection
+// of the value, combining any samples with duplicate offsets. This method is
+// the single-value equivalent of MergeTimeSeriesValues, and is used in the case
+// where the first value is merged into the key. Returns true if the merge is
+// successful.
+bool ConsolidateTimeSeriesValue(cockroach::proto::Value *val, rocksdb::Logger* logger) {
+    // Attempt to parse TimeSeriesData from both Values.
+    cockroach::proto::InternalTimeSeriesData val_ts;
+    if (!val_ts.ParseFromString(val->bytes())) {
+        rocksdb::Warn(logger,
+                "InternalTimeSeriesData could not be parsed from bytes.");
+        return false;
+    }
+
+    // Initialize new_ts and its primitive data fields.
+    cockroach::proto::InternalTimeSeriesData new_ts;
+    new_ts.set_start_timestamp_nanos(val_ts.start_timestamp_nanos());
+    new_ts.set_sample_duration_nanos(val_ts.sample_duration_nanos());
+
+    // Sort values in the ts value. 
+    std::sort(val_ts.mutable_samples()->pointer_begin(),
+            val_ts.mutable_samples()->pointer_end(),
+            TimeSeriesSampleOrdering);
```

Comment:
Not part of this change, but you should use a functor object instead of a function pointer for `TimeSeriesSampleOrdering`. In my past experience functor objects inline better (might not be true of recent compilers). Something like:

```
struct TimeSeriesSampleOrdering {
  bool operator()(const cockroach::proto::InternalTimeSeriesSample* a, const cockroach::proto::InternalTimeSeriesSample* b) const {
    return a->offset() < b->offset();
  }
};
```